core/src/navigation/services/context-switcher.js

```diff
@@ -118,6 +119,18 @@ export const ContextSwitcherHelpers = {
     return selectedLabel || (await ContextSwitcherHelpers.getFallbackLabel(fallbackLabelResolver, selectedId));
   },
 
+  async getSelectedNode(currentPath, options, parentNodePath) {
+    const selectedId = this.getSelectedId(currentPath, options, parentNodePath);
+    if (!selectedId) {
+      return;
+    }
+
+    let selectedOption = await this.getSelectedOption(currentPath, options, parentNodePath);
+    let selectedNodePath = selectedOption ? selectedOption.link : undefined;
```

Comment:
```suggestion
    const selectedOption = await this.getSelectedOption(currentPath, options, parentNodePath);
    const selectedNodePath = selectedOption ? selectedOption.link : undefined;
```
looks like `const` might be more suitable, since it's not being changed afterwards.

Reply:
It changes every time the user picks another node from the list.